broker/service/api/v10.py

```diff
@@ -153,3 +163,378 @@ def submission_log(submission_id):
+        if not authorization['success']:
+            API_LOG.log("Unauthorized request")
+            raise ex.UnauthorizedException()
+    
+    try:
+        controller.delete_cluster(api.controller_url, cluster_name, data)
+        monitor.delete_cluster(api.monitor_url, cluster_name, data)
+        visualizer.delete_cluster(api.visualizer_url, cluster_name, data)
+    except Exception:
+        API_LOG.log("Error while deleting the cluster in the other components")
+
+    conf_name = cluster_name
+
+    if(not os.path.isfile("./data/clusters/%s/%s" % (conf_name, conf_name))):
+        status = "failed"
+        reason = "cluster does not exists in this Asperathos section"
+    else:
+        # Check if the cluster to be deleted is the currently active
+        # if True, empty the config file currently being used.
+        if(filecmp.cmp("./data/clusters/%s/%s" % (conf_name, conf_name), api.k8s_conf_path)):
+            open(api.k8s_conf_path, 'w').close()
+
+        shutil.rmtree("./data/clusters/%s/" % (conf_name))
+
+        status = "success"
+        reason = ""
+
+    return {"cluster_name": conf_name, "status": status, "reason": reason}
+
+""" Activate a cluster to be used in a Asperathos section
+
+Raises:
+    ex.BadRequestException -- Missing parameters in request
+    ex.UnauthorizedException -- Authetication problem
+
+Returns:
+    dict -- Returns a dict with the cluster_name, 
+    the status of the activation (success or failed) and a
+    reason in case of 'failed' status
+"""
+
+def activate_cluster(cluster_name, data):
+
+    if ('enable_auth' not in data):
+        API_LOG.log("Missing parameters in request")
+        raise ex.BadRequestException()
+
+    enable_auth = data['enable_auth'] 
+
+    if enable_auth:
+        if 'username' not in data or 'password' not in data:
+            API_LOG.log("Missing parameters in request")
+            raise ex.BadRequestException()
+
+        username = data['username']
+        password = data['password']
+
+        authorization = authorizer.get_authorization(api.authorization_url,
+                                                 username, password)
+                                            
+        if not authorization['success']:
+            API_LOG.log("Unauthorized request")
+            raise ex.UnauthorizedException()
+
+    try:
+        controller.active_cluster(api.controller_url, cluster_name, data)
+        monitor.active_cluster(api.monitor_url, cluster_name, data)
+        visualizer.active_cluster(api.visualizer_url, cluster_name, data)
+    except Exception:
+        API_LOG.log("Error while activating the cluster into other components")
+
+    conf_name = cluster_name
+
+    if(not os.path.isfile("./data/clusters/%s/%s" % (conf_name, conf_name))):
+        status = "failed"
+        reason = "cluster does not exists in this Asperathos section"
+    else:
+
+        with open("./data/clusters/%s/%s" % (conf_name, conf_name), 'r') as f:
```

Comment:
can't you just do a `shutil.copyfile()`?

Reply:
Modification made in commit [540b521](https://github.com/ufcg-lsd/asperathos-manager/commit/540b521953feea4e73758f6d8173a1c58e9e65be)